ioctl/newcmd/bc/bcbucket.go

```diff
@@ -0,0 +1,275 @@
+		Endpoint: endpoint,
+		Insecure: insecure,
+	})
+	if err != nil {
+		return nil, err
+	}
+	method := &iotexapi.ReadStakingDataMethod{
+		Method: iotexapi.ReadStakingDataMethod_BUCKETS_BY_INDEXES,
+	}
+	methodData, err := proto.Marshal(method)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to marshal read staking data method")
+	}
+	readStakingdataRequest := &iotexapi.ReadStakingDataRequest{
+		Request: &iotexapi.ReadStakingDataRequest_BucketsByIndexes{
+			BucketsByIndexes: &iotexapi.ReadStakingDataRequest_VoteBucketsByIndexes{
+				Index: []uint64{index},
+			},
+		},
+	}
+	requestData, err := proto.Marshal(readStakingdataRequest)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to marshal read staking data request")
+	}
+
+	request := &iotexapi.ReadStateRequest{
+		ProtocolID: []byte("staking"),
+		MethodName: methodData,
+		Arguments:  [][]byte{requestData},
+	}
+
+	ctx := context.Background()
+	jwtMD, err := util.JwtAuth()
+	if err == nil {
+		ctx = metautils.NiceMD(jwtMD).ToOutgoing(ctx)
+	}
+
+	response, err := apiClient.ReadState(ctx, request)
+	if err != nil {
+		sta, ok := status.FromError(err)
+		if ok {
+			return nil, errors.New(sta.Message())
+		}
+		return nil, errors.Wrap(err, "failed to invoke ReadState api")
+	}
+	buckets := iotextypes.VoteBucketList{}
+	if err := proto.Unmarshal(response.Data, &buckets); err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal response")
+	}
+	if len(buckets.GetBuckets()) == 0 {
+		return nil, errors.New("zero len response")
+	}
+	return buckets.GetBuckets()[0], nil
+}
+
+func getBucketsCount(c ioctl.Client) (count *iotextypes.BucketsCount, err error) {
+	var endpoint string
+	var insecure bool
+
+	apiClient, err := c.APIServiceClient(ioctl.APIServiceConfig{
+		Endpoint: endpoint,
+		Insecure: insecure,
+	})
+	if err != nil {
+		return nil, err
+	}
+	method := &iotexapi.ReadStakingDataMethod{
+		Method: iotexapi.ReadStakingDataMethod_BUCKETS_COUNT,
+	}
+	methodData, err := proto.Marshal(method)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to marshal read staking data method")
+	}
+	readStakingdataRequest := &iotexapi.ReadStakingDataRequest{
+		Request: &iotexapi.ReadStakingDataRequest_BucketsCount_{
+			BucketsCount: &iotexapi.ReadStakingDataRequest_BucketsCount{},
+		},
+	}
+	requestData, err := proto.Marshal(readStakingdataRequest)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to marshal read staking data request")
+	}
+
+	request := &iotexapi.ReadStateRequest{
+		ProtocolID: []byte("staking"),
+		MethodName: methodData,
+		Arguments:  [][]byte{requestData},
+	}
+
+	ctx := context.Background()
+	jwtMD, err := util.JwtAuth()
+	if err == nil {
+		ctx = metautils.NiceMD(jwtMD).ToOutgoing(ctx)
+	}
+
+	response, err := apiClient.ReadState(ctx, request)
+	if err != nil {
+		sta, ok := status.FromError(err)
+		if ok {
+			return nil, errors.New(sta.Message())
+		}
+		return nil, errors.Wrap(err, "failed to invoke ReadState api")
+	}
```

Comment:
`getBucketByIndex` and `getBucketsCount` have the same process. and can combine together for the common code.  you can move the common process to a new function.  the different part like request(`method`, `readStakingdataRequest` ) and response is handled in each function.